### DISCUSSION TITLE
Sqlx with SeaORM in a single transaction?

### DISCUSSION THREAD

**Discussion by Adhalianna** [SeaQL/sea-orm#1134] [Ideas]
I really, really wanted to use SeaORM with sqlx together and while I found partial solution to using a single db handle in my code with [`SqlxPostgresConnector::from_sqlx_postgres_pool`](https://docs.rs/sea-orm/0.9.3/sea_orm/struct.SqlxPostgresConnector.html#method.from_sqlx_postgres_pool) the solution fell apart when I realized I need to run both sqlx and sea_orm in a single transaction. I have noticed that there is some work on [an issue, #398](https://github.com/SeaQL/sea-orm/issues/398), that seems somewhat related to tighter integration of the two but it looks like bare beginnings of such integration to me. 

Is it even possible for SeaORM to allow the following sometime in the future?
```rs
let tx = &db.begin().await.unwrap();

for r in request_elements {
  let _ = some_model.insert(&tx).await;
  
  if some_condition(r) {
    let _ = sqlx::query!("SELECT EXISTS(SELECT 1 FROM ...)").fetch_one(&tx).await;
    //...
  } else {
    let _ = other_model.insert(&tx).await;
    //...
  }
}
```
It would be amazing if something like this was on a road-map but I can easily imagine it would take at least months to get there.

Personally, I have tried to create new type wrappers around sqlx connection-related types which would implement `sea_orm::ConnectionTrait` and `sea_orm::TransactionTrait` starting from `sqlx::Pool` but I run into problems with creating a wrapper around `sqlx::Transaction`.  I have also tried briefly to fork sea_orm and implement traits like `sqlx::Acquire` on `sea_orm::DatabaseConnection` but the type puzzle felt even much more unsolvable there. 

I wanted to use both sqlx and sea_orm because I need my updates and inserts to be dynamic but my select queries are usually quite complex and expressing them with rust code (sea_query) adds unnecessary mental overhead. Having those complex queries be compile time checked would be amazing.

**Comment by billy1624** [comment]
Hey @Adhalianna, welcome!! Thanks for the report! Why not querying with SeaORM raw query interface? 
https://www.sea-ql.org/SeaORM/docs/basic-crud/raw-sql/

```rs
let _ = sqlx::query!("SELECT EXISTS(SELECT 1 FROM ...)").fetch_one(&tx).await;
```

Becomes

```rs
let query_res: Option<QueryResult> = db
    .query_one(Statement::from_string(
        DatabaseBackend::MySql,
        "SELECT EXISTS(SELECT 1 FROM ...) as `exists`".to_owned(),
    ))
    .await?;
let query_res = query_res.unwrap();
let exists: bool = query_res.try_get("", "exists")?;
```
